src/Psl/Xml/Error/XmlErrorCollection.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Xml\Error;
+
+use Countable;
+use IteratorAggregate;
+use Psl\Arr;
+use Psl\Iter;
+use Psl\Iter\Iterator;
+use Psl\Str;
+
+use const Psl\Xml\ERR_WARNING;
+
+final class XmlErrorCollection implements Countable, IteratorAggregate
+{
+    /**
+     * @var list<XmlError>
+     */
+    private array $errors;
+
+    public function __construct(XmlError ...$errors)
+    {
+        $this->errors = $errors;
+    }
+
+    /**
+     * @return Iterator<int, XmlError>
+     */
+    public function getIterator(): Iterator
+    {
+        return Iterator::create($this->errors);
+    }
+
+    public function count(): int
+    {
+        return count($this->errors);
+    }
+
+    public function ignoreWarnings(): self
+    {
+        return new self(
+            ...Arr\filter(
+                $this->errors,
+                static fn (XmlError $error): bool => $error->level() !== ERR_WARNING
+            )
+        );
+    }
```

Comment:
I think this should be moved to the constructor instead. ( see above )

Reply:
I don't agree : The collection is being filled by the detect function and handled by the implementing function.
So you don't know if you want to ignore warnings or not.

For XSD, you want to show them.
For loading documents, you don't really care about them - since they are non blocking.

I'dd like to keep it as a separate method if you don't mind.

Reply:
hm, i don't really like this method, what about `filter(...$levels): self`? that makes more sense if you want to seperate levels.

```
$all = ...;
$fatal_errors = $all->filter('fatal');
$errors = $all->filter('error');
$warnings = $all->filter('warnings');

$important = $all->filter('fatal', 'error');
```